incubator/external-dns/templates/deployment.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels: {{ include "labels" . | indent 4 }}
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels: {{ include "labels" . | indent 8 }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{- if .Values.image.repository }}{{ .Values.image.repository }}/{{- end -}}{{.Values.image.name}}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args:
+        - --provider={{ .Values.provider }}
+        - --policy={{ .Values.policy }}
+        - --registry={{ .Values.registry }}
+        - --txt-owner-id={{ .Values.txtOwnerId }}
+        {{- if .Values.compatibility }}
+        - --compatibility={{ .Values.compatibility }}
```

Comment:
I think it is possible to not guard this. We could then uncomment `compatibility:` in values.yaml. This would lead to something like:

```yaml
args:
- --compatibility=
```

in the redenred manifest when no compatibility is used, which does work as expected (no compatibility). This approach works for other flags as well, e.g. `--namespace` (see https://github.com/kubernetes/charts/blob/master/stable/chaoskube/values.yaml#L23).

However, it clutters the manifest a bit but would cleanup the template and values.yaml. Not sure about this, let me know what you think @rk295.